twine/cli.py

```diff
@@ -12,6 +12,10 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 import argparse
+from typing import Any
+from typing import Dict
+from typing import List
+from typing import Tuple
```

Comment:
Definitely a nit, but is there a reason why we're importing like this and not:

```suggestion
from typing import Any, Dict, List, Tuple
```

Reply:
Also, would be nice to avoid these imports at runtime, see what the `packaging` library is doing here: https://github.com/pypa/packaging/blob/61672bf9f507f38e84ce2786a1c42f55fa0a3153/packaging/_typing.py

Reply:
> Definitely a nit, but is there a reason why we're importing like this

That's a side-effect of:

https://github.com/pypa/twine/blob/2c30b1e9c7936a6edd2e54c44852b58461fbc771/.isort.cfg#L2-L3

which was added in #576. Still waiting on an isort release that includes https://github.com/timothycrosley/isort/pull/1085.

> would be nice to avoid these imports at runtime

I've seen the `if TYPE_CHECKING` pattern before, but never investigated why it was useful. From the `packaging` link (emphasis mine):

> Generally, `typing` would be imported at runtime and used in that fashion -
it acts as a no-op at runtime and **does not have any run-time overhead** by
design. As it turns out, `typing` uses separate sources for Python 2/Python 3. **To work around this**, mypy allows the `typing` import to be behind a False-y optional.

My read on this is that `if TYPE_CHECKING` is useful for Py2/3  compatibility, which Twine doesn't need. The mypy docs also include examples of using it for troubleshooting, e.g. [import cycles](https://mypy.readthedocs.io/en/stable/common_issues.html?highlight=type_checking#import-cycles).

So, I'm happy to add it if it's valuable, but it's not clear to me how it would be.

Reply:
Okay, so if I understand this correctly, as per the [google style guide](https://github.com/google/styleguide/blob/gh-pages/pyguide.md#31912-imports-for-typing) , we are allowed to import multiple specific classes on one line from the typing module.

But there is currently no support from isort in the version we are using to globally force single line imports, but exclude certain modules from that rule, which is what `single_line_exclusions=typing` would do in a newer version of isort once https://github.com/timothycrosley/isort/pull/1085 is merged

Reply:
@deveshks That's correct.

Reply:
@di I'm going to merge this as-is, but I'd love to hear your thoughts about `TYPE_CHECKING`.

Also: I really enjoyed your [Static Typing](https://www.youtube.com/watch?v=ST33zDM9vOE) talk.